src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts

```diff
@@ -0,0 +1,242 @@
+			this._placeholderContainer.style.visibility = 'visible';
+			this._placeholderInfoBox.text = 'Add execution plans to compare';
+		} else if (this._plans.length === 1) {
+			this._placeholderContainer.style.visibility = 'visible';
+			this._placeholderInfoBox.text = 'Add execution plans to compare';
+		} else {
+			this._placeholderContainer.style.display = 'none';
+			this._addExecutionPlanAction.enabled = false;
+		}
+	}
+
+	public async addExecutionPlan(plan: azdata.executionPlan.ExecutionPlanGraphInfo): Promise<void> {
+
+		try {
+			let executionPlanGraph = await this.executionPlanService.getExecutionPlan(plan);
+			this._plans.push(plan);
+			const planView = this._instantiationService.createInstance(ExecutionPlanComparisonView, this._planContainer, this._placeholderContainer);
+			this._planControls.push(planView);
+			this.updatePlaceHolderContainer();
+			const l = this._plans.length;
+			planView._onCellSelectedEvent(e => {
+				if (l === 1) {
+					this._propertiesView.setTopElement(e);
+				} else {
+					this._propertiesView.setBottomElement(e);
+				}
+			});
+			this._executionPlans.push(executionPlanGraph.graphs);
+			planView.addGraphs(executionPlanGraph.graphs);
+		} catch (e) {
+			this._notificationService.error(e);
+		}
+	}
+
+	public togglePropertiesView() {
+		if (this._propertiesContainer.style.display === 'none') {
+			this._propertiesContainer.style.display = '';
+
+		} else {
+			this._propertiesContainer.style.display = 'none';
+
+		}
+
+	}
+
+	public zoomIn() {
+		this._planControls.forEach(pc => {
+			pc.azdataGraphDiagram.forEach(d => {
+				d.zoomIn();
+			});
+		});
+	}
+	public zoomOut() {
+		this._planControls.forEach(pc => {
+			pc.azdataGraphDiagram.forEach(d => {
+				d.zoomOut();
+			});
+		});
+	}
+
+	public zoomToFit() {
+		this._planControls.forEach(pc => {
+			pc.azdataGraphDiagram.forEach(d => {
+				d.zoomToFit();
+			});
+		});
+	}
+}
+
+class AddExecutionPlanAction extends Action {
+	public static ID = 'ep.AddExecutionPlan';
+	public static LABEL = localize('addExecutionPlanLabel', "Add execution plan");
+	constructor(
+	) {
+		super(AddExecutionPlanAction.ID, AddExecutionPlanAction.LABEL, addIconClassName);
+	}
+
+	public override async run(context: ExecutionPlanComparisonEditorView): Promise<void> {
+		const openFileURI = await context.fileDialogService.showOpenDialog({
+			filters: [
+				{
+					extensions: await context.executionPlanService.getAllSupportedExecutionPlanExtensions(),
+					name: localize('executionPlan.SaveFileDescription', "Execution Plan Files")
+				}
+			],
+			canSelectMany: false
+		});
+
+		if (openFileURI?.length === 1) {
+			const selectedFile = openFileURI[0];
+			const selectedFileContent = (await context.textFileService.read(selectedFile, { acceptTextOnly: true })).value;
+			context.addExecutionPlan({
+				graphFileContent: selectedFileContent,
+				graphFileType: extname(selectedFile.fsPath).replace('.', '')
+			});
+
+		} else {
+			// no file selected.
```

Comment:
This `else` block seems unnecessary since we're not doing anything with it. Is there a reason we're keeping it around?